azext_iot/iothub/providers/message_endpoint.py

```diff
@@ -224,6 +218,226 @@ def create(
+                if not endpoint_subscription_id:
+                    endpoint_subscription_id = self.hub_resource.additional_properties['subscriptionid']
+            elif EndpointType.CosmosDBContainer.value != endpoint_type.lower():
+                original_endpoint.connection_string = connection_string
+            else:
+                if primary_key:
+                    original_endpoint.primary_key = primary_key
+                if secondary_key:
+                    original_endpoint.secondary_key = secondary_key
+
+        # Properties + connection string by specific types
+        if fetch_connection_string:
+            if EndpointType.EventHub.value == endpoint_type.lower():
+                original_endpoint.connection_string = get_eventhub_cstring(
+                    cmd=self.cli,
+                    namespace_name=endpoint_account_name,
+                    eventhub_name=entity_path,
+                    policy_name=endpoint_policy_name,
+                    rg=endpoint_resource_group,
+                    sub=endpoint_subscription_id
+                )
+            elif EndpointType.ServiceBusQueue.value == endpoint_type.lower():
+                original_endpoint.connection_string = get_servicebus_queue_cstring(
+                    cmd=self.cli,
+                    namespace_name=endpoint_account_name,
+                    queue_name=entity_path,
+                    policy_name=endpoint_policy_name,
+                    rg=endpoint_resource_group,
+                    sub=endpoint_subscription_id
+                )
+            elif EndpointType.ServiceBusTopic.value == endpoint_type.lower():
+                original_endpoint.connection_string = get_servicebus_topic_cstring(
+                    cmd=self.cli,
+                    namespace_name=endpoint_account_name,
+                    topic_name=entity_path,
+                    policy_name=endpoint_policy_name,
+                    rg=endpoint_resource_group,
+                    sub=endpoint_subscription_id
+                )
+        elif endpoint_type in [
+            EndpointType.EventHub.value, EndpointType.ServiceBusQueue.value, EndpointType.ServiceBusTopic.value
+        ] and entity_path and not connection_string:
+            # only set entity_path if no connection string
+            original_endpoint.entity_path = entity_path
+
+        if endpoint_type == EndpointType.AzureStorageContainer.value:
+            if fetch_connection_string:
+                original_endpoint.connection_string = get_storage_cstring(
+                    cmd=self.cli,
+                    account_name=endpoint_account_name,
+                    rg=endpoint_resource_group,
+                    sub=endpoint_subscription_id
+                )
+            if container_name:
+                original_endpoint.container_name = container_name
+            if encoding:
+                original_endpoint.encoding = encoding.lower()
+            if file_name_format:
```

Comment:
was originally thinking of adding the option to revert to default here but decided against since it creates more custom param inputs (like "update" or "")

Reply:
Most of our control-plane updates (checkout CLI core for legacy examples 🙃) use the format:

```
def update():
   model = get_model()
   
   if new_property_value:
      model.old_property = new_property_value
   if other_new_property_value:
      model.other_old_property = other_new_property_value
      
   service.update(model)
```

I think it's totally acceptable to simply "only update the params that are passed in" as it matches our existing patterns.
Future thinking / brainstorming out loud - maybe there's a possibility to add a `--set-default prop1 prop2` type command which takes in a space-separated list of properties to reset to their default values? I can see this being very tricky to scope / troubleshoot though.